azure-servicebus/azure/servicebus/aio/async_client.py

```diff
@@ -0,0 +1,654 @@
+            'skip': types.AMQPInt(skip),
+            'top': types.AMQPInt(max_results),
+        }
+        async with BaseHandler(
+                self.loop, self.entity_uri, self.auth_config, debug=self.debug, **kwargs) as handler:
+            return await handler._mgmt_request_response(  # pylint: disable=protected-access
+                REQUEST_RESPONSE_GET_MESSAGE_SESSIONS_OPERATION,
+                message,
+                mgmt_handlers.list_sessions_op)
+
+    def get_receiver(self, session=None, prefetch=0, mode=ReceiveSettleMode.PeekLock, idle_timeout=0, **kwargs):
+        """Get a Receiver for the ServiceBus endpoint. A Receiver represents
+        a single open Connection with which multiple receive operations can be made.
+        This operation is only available on Python 3.5 and above.
+
+        :param session: A specific session from which to receive. This must be specified for a
+         sessionful entity, otherwise it must be None. In order to receive the next available
+         session, set this to NEXT_AVAILABLE.
+        :type session: str or ~azure.servicebus.NEXT_AVAILABLE
+        :param prefetch: The maximum number of messages to cache with each request to the service.
+         The default value is 0, i.e. messages will be received from the service and processed
+         one at a time. Increasing this value will improve message through-put performance but increase
+         the change that messages will expire while they are cached if they're not processed fast enough.
+        :type prefetch: int
+        :param mode: The mode with which messages will be retrieved from the entity. The two options
+         are PeekLock and ReceiveAndDelete. Messages received with PeekLock must be settled within a given
+         lock period before they will be removed from the queue. Messages received with ReceiveAndDelete
+         will be immediately removed from the queue, and cannot be subsequently rejected or re-received if
+         the client fails to process the message. The default mode is PeekLock.
+        :type mode: ~azure.servicebus.common.constants.ReceiveSettleMode or str
+        :param idle_timeout: The timeout in seconds between received messages after which the receiver will
+         automatically shutdown. The default value is 0, i.e. no timeout.
+        :type idle_timeout: int
+        :returns: A Receiver instance with an unopened Connection.
+        :rtype: ~azure.servicebus.receive_handler.Receiver
+        :raises: If the current Service Bus entity requires sessions, a TypeError will
+         be raised.
+        """
+        if self.entity and not self.requires_session and session:
+            raise ValueError("A session cannot be used with a non-sessionful entitiy.")
+        if self.entity and self.requires_session and not session:
+            raise ValueError("This entity requires a session.")
+        assert prefetch >= 0, "Prefetch must be an integer between 0 and 50000 inclusive."
```

Comment:
The assert doesn't match what you are testing.
Being that assertion can be disabled with optimized compilation of pyc, if it's important check should be a "if/raise"